e2e/kube/pod.go

```diff
@@ -24,13 +24,16 @@ import (
 	corev1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/fields"
+	"k8s.io/apimachinery/pkg/runtime"
 	"k8s.io/client-go/kubernetes"
 	"k8s.io/client-go/rest"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
 // PodLogs returns the logs of the pod with the given name.
 func PodLogs(ctx context.Context, restConfig *rest.Config, namespace, name, container string) (string, error) {
+	coreConfig := rest.CopyConfig(restConfig)
```

Comment:
do we need to copy?

Reply:
I think it's safer - without a copy, we would change `restConfig` (passed by pointer here) that might be passed somewhere later to a dynamic clientset and CRs do not support protobuf encoding, unfortunately.